helios-client/src/main/java/com/spotify/helios/common/descriptors/SecretVolumeRequest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.common.descriptors;
+
+import static java.util.Collections.emptyMap;
+
+import com.spotify.helios.common.SecretVolumeException;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a secret volume request to secret-volume, i.e. a secret-volume response with an added
+ * KeyPair.
+ *
+ * <pre>
+ *   {
+ *     "ID": "containerId",
+ *     "Source": "Talos",
+ *     "Tags": {},
+ *     "KeyPair": {
+ *        "Certificate": "PEM encoded cert",
+ *        "PrivateKey": "Pem encoded private key"
+ *     }
+ *   }
+ * </pre>
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SecretVolumeRequest extends SecretVolume {
+
+  private final KeyPair keyPair;
+
+  public SecretVolumeRequest(
+      @JsonProperty("ID") final String id,
+      @JsonProperty("Source") final Secrets.Source source,
+      @JsonProperty("Tags") final Map<String, List<String>> tags,
+      @JsonProperty("KeyPair") final KeyPair keyPair) {
+    super(id, source, tags);
+    this.keyPair = keyPair;
+  }
+
+  public static SecretVolumeRequest create(final String id, final Secrets.Source source,
+                                           final String certData, final String keyData) {
+    // TODO(negz): Read tags from config, or agent labels?
+    final Map<String, List<String>> tags = emptyMap();
+    return new SecretVolumeRequest(id, source, tags, new KeyPair(certData, keyData));
+  }
+
+  private static String readToString(final String path) throws SecretVolumeException {
+    try {
+      return new String(Files.readAllBytes(Paths.get(path)), Charset.defaultCharset());
```

Comment:
this seems to be the same exact code as `class FsSmallFileReader` - why does it need to be a static method too? 

I don't think this method is used in the class